react/week5/github-search-router/github-search-router/src/components/Form.jsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import TextField from '@material-ui/core/TextField';
+import { UserContext } from '../context/Context';
+
+export default class Form extends React.Component {
+   render() {
+      return (
+         <UserContext.Consumer>
+            {(value) => {
+               const { handleChange, findUser, userName } = value;
+               return (
+                  <div>
+                     <h1>Github Search</h1>
+                     <form onSubmit={findUser} noValidate autoComplete='off'>
+                        <TextField
+                           id='outlined-secondary'
+                           label='Search users'
+                           variant='outlined'
+                           value={userName}
```

Comment:
When it comes to setting the state and controlling the value of input elements, I would generally recommend keeping the input state within the same component you're using the input. It can lead to unnecessary indirection.

Reply:
I would usually structure my code so that the `findUser` function would take the state of the inputs directly in this component as arguments. 

I think it's clever to expose the onsubmit function through context, good work with that!